src/Tooltip/index.tsx

```diff
@@ -0,0 +1,178 @@
+import React from 'react'
+import {useSSRSafeId} from '@react-aria/ssr'
+import type {AnchorPosition, AnchorSide, AnchorAlignment} from '@primer/behaviors'
+import Box from '../Box'
+import {useAnchoredPosition} from '../hooks'
+import {SxProp, merge, BetterSystemStyleObject} from '../sx'
+
+type TooltipDirection = 'nw' | 'n' | 'ne' | 'e' | 'se' | 's' | 'sw' | 'w'
+type TooltipAlign = 'left' | 'right'
+
+export type TooltipProps = {
+  /** The text content of the tooltip. This should be brief and no longer than a sentence.
+   *  Marked as optional to support backward compatibility with aria-label. */
+  text?: string
+  /** @deprecated Use `text` instead */
+  'aria-label'?: string
+  /** Direction relative to target */
+  direction?: TooltipDirection
+  /** @deprecated Use `direction` instead. Alignment relative to target. */
+  align?: TooltipAlign
+  /** Use aria-describedby or aria-labelledby */
+  type?: 'description' | 'label'
+  /** Tooltip target */
+  children: React.ReactElement
+  /** When set to true, tooltip appears without any delay */
+  noDelay?: boolean
+  /** @deprecated Always set to true now. */
+  wrap?: boolean
+} & SxProp
+
+// map tooltip direction to anchoredPosition props
+const directionToPosition: Record<TooltipDirection, {side: AnchorSide; align: AnchorAlignment}> = {
+  nw: {side: 'outside-top', align: 'start'},
+  n: {side: 'outside-top', align: 'center'},
+  ne: {side: 'outside-top', align: 'end'},
+  e: {side: 'outside-right', align: 'center'},
+  se: {side: 'outside-bottom', align: 'end'},
+  s: {side: 'outside-bottom', align: 'center'},
+  sw: {side: 'outside-bottom', align: 'start'},
+  w: {side: 'outside-left', align: 'center'}
+}
+
+// map align to AnchorAlignment
+const alignToAnchorAlignment: Record<TooltipAlign, AnchorAlignment> = {left: 'start', right: 'end'}
+
+export const TooltipContext = React.createContext<{tooltipId?: string}>({})
+
+export const Tooltip: React.FC<TooltipProps> = ({
```

Comment:
notes for reviewer: This is a breaking change, Tooltip used to export as default. But, looks like it's safe because no one is using it from the direct path (`@primer/react/lib-esm/Tooltip`): https://cs.github.com/?scopeName=All+repos&scope=&q=%40primer%2Freact+Tooltip